transfer_queue/client.py

```
@@ -293,12 +293,12 @@ async def async_put(
             )
 
         if metadata is None:
-            assert global_step is not None, "global_steps must be provided if metadata is not given"
+            assert partition_id is not None, "partition_id must be provided if metadata is not given"
```

better to raise an error. assert might be ignored